src/tools/pkcs11-tool.c

```diff
@@ -1896,6 +1903,14 @@ get_uri(CK_TOKEN_INFO_PTR info)
 	const char *serial = percent_encode(info->serialNumber, sizeof(info->serialNumber));
 	copy_key_value_to_uri("serial=", serial, CK_FALSE);
 	const char *token = percent_encode(info->label, sizeof(info->label));
+
+	if (opt_uri_with_slot_id) {
+		copy_key_value_to_uri("token=", token, CK_FALSE);
+		static char slot_id_str[32];
+		snprintf(slot_id_str, sizeof(slot_id_str), "%lu", slot);
+		return copy_key_value_to_uri("slot-id=", slot_id_str, CK_TRUE);
+	}
+
 	return copy_key_value_to_uri("token=", token, CK_TRUE);
```

Comment:
the order of the parameters in the uri really does not matter so I would rather move it to the beginning than duplicating the adding the token label in two places. It makes it hard to follow and understand.
```suggestion

	if (opt_uri_with_slot_id) {
		static char slot_id_str[32];
		snprintf(slot_id_str, sizeof(slot_id_str), "%lu", slot);
		copy_key_value_to_uri("slot-id=", slot_id_str, CK_FALSE);
	}

	const char *token = percent_encode(info->label, sizeof(info->label));
	return copy_key_value_to_uri("token=", token, CK_TRUE);
```
(or even before all the other parameters)